pkg/object/graph/graph.go

```diff
@@ -141,3 +145,18 @@ func (g *Graph) Sort() ([][]object.ObjMetadata, error) {
 	}
 	return sorted, nil
 }
+
+// CyclicDependencyError when directed acyclic graph contains a cycle.
+// The cycle makes it impossible to topological sort.
+type CyclicDependencyError struct {
+	Edges []Edge
+}
+
+func (cde CyclicDependencyError) Error() string {
+	var errorBuf bytes.Buffer
+	errorBuf.WriteString("cyclic dependency")
+	for _, edge := range cde.Edges {
+		errorBuf.WriteString(fmt.Sprintf("\n\t%s -> %s", edge.From, edge.To))
```

Comment:
Do we have any other examples where we print the ObjMetadata for resources? The example in the PR description seems to suggest a format that I don't think we have used elsewhere.

Reply:
I have updated the printing to the more standard: <NAMESPACE>/<NAME>. It now looks like
```
error: cyclic dependency
	test-cyclic-error-ns/pod-b -> test-cyclic-error-ns/pod-c
	test-cyclic-error-ns/pod-c -> test-cyclic-error-ns/pod-a
	test-cyclic-error-ns/pod-a -> test-cyclic-error-ns/pod-b
```